android/src/main/kotlin/com/mopinion/mopinion_flutter_integration_plugin/MopinionFlutterIntegrationPlugin.kt

```diff
@@ -26,103 +26,151 @@ import io.flutter.plugin.common.MethodChannel.Result
-  private lateinit var activity: Activity
-  private lateinit var mopinion: Mopinion
-  private lateinit var eventChannel: EventChannel
-
-  private var eventSink: EventChannel.EventSink? = null
-
-  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
-    channel = MethodChannel(flutterPluginBinding.binaryMessenger, CHANNEL)
-    channel.setMethodCallHandler(this)
-    eventChannel = EventChannel(flutterPluginBinding.binaryMessenger, EVENT_CHANNEL_NAME)
-    eventChannel.setStreamHandler(this)
-  }
-
-  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
-    registry = binding.textureRegistry
-
-    channel.setMethodCallHandler(null)
-  }
-
-  override fun onMethodCall(call: MethodCall, result: Result) {
-    when(MopinionActions.map[call.method]) {
-      MopinionActions.InitialiseSDK -> {
-        val deploymentKey = call.argument(DEPLOYMENT_KEY) as String? ?: return
-        val version = call.argument(VERSION) as String? ?: return
-        val isLogActive = call.argument(LOG) as Boolean? ?: return
-        Mopinion.initialiseFromFlutter(
-          application = activity.application,
-          deploymentKey = deploymentKey,
-          pluginVersion = version,
-          log = isLogActive
-        )
-      }
-      MopinionActions.TriggerEvent -> {
-        val eventName = call.argument(FIRST_ARGUMENT) as String? ?: return
-        mopinion = Mopinion(activity as FlutterFragmentActivity, activity as FlutterFragmentActivity)
-        mopinion.event(eventName) { formState ->
-          if (formState is FormState.Loading) {
-            if (formState.isLoading) {
-              Log.d("FlutterFragmentActivity", "Loading")
-              eventSink?.success("Loading")
-            } else {
-              Log.d("FlutterFragmentActivity", "NotLoading")
-              eventSink?.success("NotLoading")
+class MopinionFlutterIntegrationPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
+    EventChannel.StreamHandler {
+
+    private lateinit var channel: MethodChannel
+    private lateinit var registry: TextureRegistry
+    private lateinit var activity: Activity
+    private lateinit var mopinion: Mopinion
+    private lateinit var eventChannel: EventChannel
+
+    private var eventSink: EventChannel.EventSink? = null
+
+    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+        channel = MethodChannel(flutterPluginBinding.binaryMessenger, CHANNEL)
+        channel.setMethodCallHandler(this)
+        eventChannel = EventChannel(flutterPluginBinding.binaryMessenger, EVENT_CHANNEL_NAME)
+        eventChannel.setStreamHandler(this)
+    }
+
+    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
+        registry = binding.textureRegistry
+
+        channel.setMethodCallHandler(null)
+    }
+
+    override fun onMethodCall(call: MethodCall, result: Result) {
```

Comment:
@adam-abbas hello Adam, thanks for contributing to our flutter plugin, appreciated.
Indeed would be appropriate to return the results, as I've seen onMethodCall doesnt returns anything I didn't return anything, but seems that under the hood (in the flutter platform) it does. Interesting!
I will merge it. 😁